src/main/java/io/jaegertracing/tests/CreateSpansRunnable.java

```diff
@@ -0,0 +1,151 @@
+                    maxDuration = 100;
+                }
+
+                delay = (maxDuration * 1000L) / config.getSpansCount();
+                logger.debug("Maximun execution time per round:{}ms, delay between spans:{}us, spans count:{}",
+                        maxDuration, delay, config.getSpansCount());
+            } else {
+                delay = config.getPerformanceTestSpanDelay() * 1000L; // delay in microseconds
+            }
+        }
+
+        @Override
+        public void run() {
+            if (config.isPerformanceTestLongRunEnabled()
+                    && executedCount.get() >= config.getPerformanceTestDuration()) {
+                return;
+            }
+            Map<String, Object> logs = new HashMap<>();
+            logs.put("event", Tags.ERROR);
+            logs.put("error.object", new RuntimeException());
+            logs.put("class", this.getClass().getName());
+            int count = 0;
+            long startTime = System.currentTimeMillis();
+            do {
+                count++;
+                // emulate client spans
+                Span span = tracer.buildSpan(name)
+                        .withTag(Tags.COMPONENT.getKey(), "perf-test")
+                        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                        .withTag(Tags.HTTP_METHOD.getKey(), "get")
+                        .withTag(Tags.HTTP_STATUS.getKey(), 200)
+                        .withTag(Tags.HTTP_URL.getKey(), "http://www.example.com/foo/bar?q=bar")
+                        .start();
+                span.log(logs);
+                span.finish();
+                try {
+                    TimeUnit.MICROSECONDS.sleep(delay);
+                } catch (InterruptedException ex) {
+                    logger.error("exception, ", ex);
+                }
+            } while (count < config.getSpansCount());
+            if (config.isPerformanceTestLongRunEnabled()) {
+                executedCount.incrementAndGet();
+                if (executedCount.get() % 60 == 0 || executedCount.get() == 1) { // print every 1 minute once
+                    logger.debug("Round number:{}, duration:{}ms, Tracer:{}",
+                            executedCount.get(), System.currentTimeMillis() - startTime, name);
+                }
+            } else {
+                logger.debug("Reporting spans done, duration:{}, Tracer:{}",
+                        TestUtils.timeTaken(System.currentTimeMillis() - startTime), name);
+            }
+        }
+
+    }
+
+    private JaegerTracer tracer;
+    private String name;
+    private TestConfig config;
+    private boolean close;
+    private AtomicInteger executedCount = new AtomicInteger(0);
+
+    private final int expectedCount;
+
+    public CreateSpansRunnable(JaegerTracer tracer, String name, TestConfig config, boolean close) {
+        this.tracer = tracer;
+        this.name = name;
+        this.config = config;
+        this.close = close;
```

Comment:
I think we should move close out of this class. This should only report spans.

Reply:
You mean `close = true` always?

Reply:
No I mean reporter should only report and not close the tracer/reporter at all.